src/tests/functional/plugin/cpu/subgraph_tests/src/seq_native_order.cpp

```diff
@@ -0,0 +1,299 @@
+public:
+    static std::string getTestCaseName(const testing::TestParamInfo<SeqParams> &obj) {
+        SEQ_TYPE seqType;
+        size_t hidden_size, input_size;
+        InputShapeParams inShapeParams;
+        std::vector<std::string> activations;
+        float clip;
+        bool linearBeforeReset;
+        ov::op::RecurrentSequenceDirection direction;
+        ElementType netPrecision;
+
+        std::tie(seqType, hidden_size, input_size, inShapeParams, activations, clip, linearBeforeReset, direction, netPrecision) = obj.param;
+
+        std::vector<ov::Dimension> bounds;
+        std::vector<TargetShapeParams> targetShapes;
+        std::tie(bounds, targetShapes) = inShapeParams;
+
+        std::ostringstream result;
+
+        if (seqType == SEQ_TYPE::GRU) {
+            result << "GRU_";
+        } else if (seqType == SEQ_TYPE::LSTM) {
+            result << "LSTM_";
+        } else if (seqType == SEQ_TYPE::RNN) {
+            result << "RNN_";
+        } else {
+            IE_THROW() << "Unsupported seq type";
+        }
+        result << "hidden_size=" << hidden_size << "_input_size=" << input_size << "_";
+        result << "batch_size_dyn=" << bounds[0] << "_seq_length_dyn=" << bounds[1] << "_";
+        for (const auto &ts : targetShapes) {
+            size_t bs, sl;
+            std::tie(bs, sl) = ts;
+            result << "(bs=" << bs << "_sl=" << sl << ")_";
+        }
+
+        result << "activations=" << CommonTestUtils::vec2str(activations)  << "_";
+        result << "clip=" << clip << "_";
+        result << "linear=" << linearBeforeReset << "_";
+        result << "direction=" << direction << "_";
+        result << "netPrec=" << netPrecision;
+
+        return result.str();
+    }
+
+protected:
+    void SetUp() override {
+        const size_t batch_size_pos = 0;
+        const size_t seq_length_pos = 1;
+
+        SEQ_TYPE seqType;
+        size_t hidden_size, input_size;
+        InputShapeParams inShapeParams;
+        std::vector<std::string> activations;
+        float clip;
+        bool linearBeforeReset;
+        ov::op::RecurrentSequenceDirection direction;
+        ElementType netPrecision;
+
+        std::tie(seqType, hidden_size, input_size, inShapeParams, activations, clip, linearBeforeReset, direction, netPrecision) = this->GetParam();
+
+        std::vector<ov::Dimension> bounds;
+        std::vector<TargetShapeParams> targetShapes;
+        std::tie(bounds, targetShapes) = inShapeParams;
+
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+
+        seqLengthInIdx = (seqType == SEQ_TYPE::LSTM ? 3 : 2);
+
+        const size_t numDirections = direction == ov::op::RecurrentSequenceDirection::BIDIRECTIONAL ? 2 : 1;
+
+        // dynamic shapes
+        ngraph::PartialShape X_shape(std::vector<ngraph::Dimension>{bounds[seq_length_pos], bounds[batch_size_pos], ngraph::Dimension(input_size)});
```

Comment:
Please use ov namespace, due to ngraph is deprecated.

Reply:
done